src/components/structures/RoomView.js

```diff
@@ -296,7 +360,7 @@ module.exports = React.createClass({
 
         if (ev.getSender() !== MatrixClientPeg.get().credentials.userId) {
             // update unread count when scrolled up
-            if (!this.state.searchResults && this.refs.messagePanel && this.refs.messagePanel.isAtBottom()) {
+            if (!this.state.searchResults && this.state.atBottom) {
```

Comment:
`this.state.atBottom` primarily controls whether the 'scroll to bottom' knob appears. It existed before this change, and is set in `onMessageListScroll` (line 626):

```javascript
         if (this.refs.messagePanel.isAtBottom() &&
                 !this._timelineWindow.canPaginate(EventTimeline.FORWARDS)) {
             // ...
             if (!this.state.atBottom) {
                 this.setState({ atBottom: true });                
             }
         }
         else {
             if (this.state.atBottom) {
                 this.setState({ atBottom: false });
             }            
         }
```

As you can see, the difference between it and `messagePanel.isAtBottom` is that it (now) also takes into account whether we can paginate forwards from our view of the timeline. The point being that, if you're looking at a historical section of timeline, you still want to increase the unread message count even if the user hits the bottom of the loaded events.